google/cloud/spanner_dbapi/connection.py

```diff
@@ -432,14 +426,13 @@ def commit(self):
         self.run_prior_DDL_statements()
         if self._spanner_transaction_started:
             try:
-                if self.read_only:
-                    self._snapshot = None
-                else:
+                if not self._read_only:
                     self._transaction.commit()
 
                 self._release_session()
                 self._statements = []
                 self._transaction_begin_marked = False
+                self._spanner_transaction_started = False
             except Aborted:
                 self.retry_transaction()
                 self.commit()
```

Comment:
Flag `_spanner_transaction_started` is being set to False at the start of every retry loop in `retry_transaction` method on line 287.
If an exception is being thrown from `retry_transaction` then `_spanner_transaction_started` might remain True which will make Connection unusable. This is the common issue across Connection/Cursor class where an uncaught exception can lead to Connection object having invalid state. So if the application programmer catches that exception and tried to use the same connection object it will lead to issues.
Should we always close the Connection in case of uncaught exception?

Reply:
Updated the code to always set these 2 flags to False post commit and rollback even if there is an uncaught exception